dfsclient/src/main/java/org/hpccsystems/dfs/client/HpccRemoteFileReader.java

```diff
@@ -416,6 +440,7 @@ private boolean retryRead()
                 this.inputStream.skip(bytesToSkip);
 
                 this.binaryRecordReader = new BinaryRecordReader(this.inputStream, resumeInfo.recordReaderStreamPos, this.readSpan);
+                this.binaryRecordReader.setRecordBuildingSpanBatchSizeKB(context.getReadRequestSpanBatchSize() * (int) context.getReadSizeKB());
```

Comment:
should we be concerned about the potential data loss in the long to int cast? if not, does getReadSizeKB need to return a long?

Reply:
I think the 32bit range should be good here, it would allow for 2TB span sizes due to counting KB not bytes. I don't see any use case for having spans representing that much data.